lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -534,10 +526,6 @@ private function isALegacyJsonComparison(Types\Type $one, Types\Type $other) : b
      */
     public function diffIndex(Index $index1, Index $index2)
     {
-        if ($index1->isFullfilledBy($index2) && $index2->isFullfilledBy($index1)) {
-            return false;
-        }
-
-        return true;
+        return ! ($index1->isFullfilledBy($index2) && $index2->isFullfilledBy($index1));
```

Comment:
I'd argue this approach is better than the original one. When debugging code, it may be convenient to set a break point on one of the `return`s and inspect the scope before switching the context. Especially, when the condition is complex. A better rework could look like:
```php
if (!$index1->isFullfilledBy($index2)) {
    return true;
}

if (!$index2->isFullfilledBy($index1)) {
    return true;
}

return false;
```

Reply:
May I do this change in another PR? Keeping one proposal per PR? I agree with you, and there are other places that we can change it :smile:

Reply:
I find this much more readable and simpler to be honest:

```php
return ! $index1->isFullfilledBy($index2) || ! $index2->isFullfilledBy($index1);
```

Reply:
@lcobucci that's just De Morgan's law applied (except that @carusogabriel's is the "simplified" version ;-) )

Reply:
> "De Morgan's law"

Learn something new today :smile: